### PR TITLE
COMPASS-3622 edit saved favourite

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Copy saved connections in the contextual menu (use `prev favorite name (copy)` as a new favorite name and no color as a default color)
- Notify a user if a favorite was changed and not saved
- Allow a user to save changes made in favorites
- Allow a user to discard changes made in favorites
- Allow user to save recents as favorites in the contextual menu (use `host:port` as a new favorite name and no color as a default color)

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/form/form-actions.jsx [hunk 1]
```diff
@@ -58,6 +79,24 @@ class FormActions extends React.Component {
...
+   * Renders a warning that a favorite was changed and changes can be saved
+   * or discarded.
+   *
+   * @returns {React.Component}
+   */
+  renderUnsavedMessage() {
+    return (
+      <div>
+        You have unsaved changes. <a
+          onClick={this.onFavoriteChangeDiscarded}
```
Comment (lrlna) [inline src/components/form/form-actions.jsx]: nit: i find this style of writing `<a>` tags super unreadable :(

File: src/stores/index.js [hunk 2]
```diff
@@ -358,6 +380,44 @@ const Store = Reflux.createStore({
...
+
+    this._addConnection(newConnection);
+  },
+
+
+  /**
+   * Discards favorite changes.
+   */
+  onFavoriteChangeDiscarded() {
+    this.state.connections.fetch({ success: () => this.trigger(this.state) });
```
Comment (lrlna) [inline src/stores/index.js]: what happens on error here?

Comment (alenakhineika) [reply src/stores/index.js]: We don't update the state the same as we did before to initialize the state. Take a look at `init()` method at the beginning of the file. This is a callback of the ampersand js if you want to add some custom logic after fetching documents.

File: src/components/form/form-actions.jsx (hunk 1 above)
Comment (alenakhineika) [reply src/components/form/form-actions.jsx]: do you have suggestions? to keep links and text at the same line.

File: src/components/form/form-actions.jsx [hunk 3]
```diff
@@ -58,6 +79,24 @@ class FormActions extends React.Component {
...
   }
 
+  /**
+   * Renders a warning that a favorite was changed and changes can be saved
+   * or discarded.
+   *
+   * @returns {React.Component}
+   */
+  renderUnsavedMessage() {
+    return (
```
[the remark this answers is not in the document]
Comment (lrlna) [reply src/components/form/form-actions.jsx]: (edited to spell favourite the same way as you have it)

**Comment by alenakhineika** [comment]
@lrlna updated!

**Comment by alenakhineika** [comment]
@lrlna updated!

**Comment by lrlna** [comment]
looks good now!

**Pull request merged**